crates/ide_assists/src/handlers/generate_function.rs

```diff
@@ -503,12 +507,25 @@ fn deduplicate_arg_names(arg_names: &mut Vec<String>) {
     }
 }
 
-fn fn_arg_name(arg_expr: &ast::Expr) -> String {
+fn fn_arg_name(ctx: &AssistContext, arg_expr: &ast::Expr) -> String {
     let name = (|| match arg_expr {
-        ast::Expr::CastExpr(cast_expr) => Some(fn_arg_name(&cast_expr.expr()?)),
+        ast::Expr::CastExpr(cast_expr) => Some(fn_arg_name(ctx, &cast_expr.expr()?)),
         expr => {
-            let s = expr.syntax().descendants().filter_map(ast::NameRef::cast).last()?.to_string();
-            Some(to_lower_snake_case(&s))
+            let name_ref = expr.syntax().descendants().filter_map(ast::NameRef::cast).last()?;
+            if let Some(NameRefClass::Definition(def)) =
+                NameRefClass::classify(&ctx.sema, &name_ref)
+            {
+                match def {
+                    Definition::ModuleDef(ModuleDef::Const(_)) => {
+                        return Some(name_ref.to_string().to_lowercase());
+                    }
+                    Definition::ModuleDef(ModuleDef::Static(_)) => {
+                        return Some(name_ref.to_string().to_lowercase());
+                    }
+                    _ => {}
+                }
+            };
```

Comment:
This can be collapsed with or patterns to something like
```rs
if let Some(NameRefClass::Definition(Definition::ModuleDef(ModuleDef::Const(_) | ModuleDef::Static(_))))) = NameRefClass::classify(&ctx.sema, &name_ref) {
    ...
}
```
Making it a `match` might be clearer though

Reply:
Yes, I think yours is better.